site/hxapi/asynchronous-bookings.md

```diff
@@ -0,0 +1,143 @@
+{% endcodetabs %}
+
+
+### Behavior  
+
+- **`true`:** Availability searches include async products.  
+- **`false`:** Availability searches exclude async products (default behavior).  
+
+---
+
+## Booking Status Polling  
+
+When working with asynchronous bookings, partners must poll our API to determine the status of a booking and when full booking details are ready.  
+
+### When Polling is Required  
+
+Polling is required when a booking has been submitted but is not yet confirmed. The `bookingStatus` field in the response from the `view booking` endpoint indicates the current status of a booking.  
+
+Booking details such as the QR code, supplier reference, and other information will only be provided once the `bookingStatus` field is in a `CONFIRMED` state.
+
+### How Often to Poll  
+
+The frequency of polling depends on the product type and supplier:  
+- **Standard Products:** Poll every 5–10 minutes.  
+- **Long Confirmation Times (e.g., specific suppliers):** Poll every 15–30 minutes.  
+
+### Booking Status Values  
+
+| Status       | Description                                                                 |
+|--------------|-----------------------------------------------------------------------------|
+| `PENDING`    | Booking has been submitted but not yet confirmed with the supplier.         |
+| `CONFIRMED`  | Booking has been successfully confirmed by the supplier. Full details are available. |
+| `FAILED`     | Booking has failed with the supplier. |
+
+---
+
+## API Endpoint  
+
+### Method  
+
+GET  
+
+### Endpoint  
+The endpoint to use is the existing view booking endpoint: [/hxapi/viewamendcancel/view/](/hxapi/viewamendcancel/view/)
+
+
+## Response Example
+
+The response will now include a `bookingStatus` field to indicate the current status of the booking.
+
+```json
+{
+  "bookingStatus": "PENDING"
+}
+```
+
+---
+
+## Best Practices  
+
+1. **Polling Interval:**  
+   - Follow the recommended intervals based on the product type or supplier.  
+   - Avoid excessive requests to prevent unnecessary load on our systems.  
+
+2. **Stop Polling:**  
+   - When the `bookingStatus` is `CONFIRMED` or `FAILED`, stop polling and take the appropriate next steps.  
+
+3. **Retry Policy:**  
+   - Use exponential backoff for retries after transient errors (e.g., 500 errors or timeouts).  
+
+---
+
+## Workflow Example  
+
+1. **Make a booking** 
+2. **Start Polling:** Use the view booking endpoint to check the `bookingStatus`.  
+   - **If `PENDING`:** Continue polling at the recommended intervals.  
+   - **If `CONFIRMED`:** Retrieve full details and stop polling.  
+   - **If `FAILED`:** Handle the failure appropriately.  
+3. **Retrieve Full Details:** Once the booking is confirmed, on the view booking response you'll have the complete booking information using the same endpoint.
+
+By following these practices, partners can ensure an optimal customer experience while integrating with asynchronous products.  
+
+---
```

Comment:
We'd also need to address:
- amendments while booking is pending
- cancellations while booking is pending

Reply:
do you mean in this iteration or on a future one? I am asking this because we lack a bit of context/ internal documentation on how we plan to address these.

Reply:
I agree, I don't think we've properly determined how these cases will work yet.

We could maybe add headings, and just say "Documentation to follow" or something? At least then it's identified as a known gap.

Reply:
I do not remember what was decided in our meeting about this. 
Do I add a "Documentation to follow for amendments and cancellations" ?